src/entrypoints/background.ts

```diff
@@ -20,4 +25,39 @@ export default defineBackground(() => {
             });
         }
     });
+
+    browser.omnibox.onInputChanged.addListener((text, suggest) => {
+        const keywords = text.toLowerCase().split(' ');
+        const suggestions = [];
+
+        for (const buffItem of Object.keys(buffItems)) {
+            if (keywords.every(k => buffItem.toLowerCase().includes(k))) {
+                suggestions.push({
+                    deletable: false,
+                    description: buffItem,
+                    content: `https://buff.163.com/goods/${buffItems[buffItem]}`,
+                });
+
+                if (suggestions.length > 5) {
+                    break;
+                }
+            }
+        }
```

Comment:
Lots of potential performance improvements here. Both execution time and memory (currently, the full buffItems object should be around 1-2 mb in memory.

For me, this whole keyword filter took around 2-8 ms, depending on the number of keywords (whitespace in the text).

Reply:
So by grouping the `buff-items-sorted.json` under common prefixes (e.g. `{"Sticker": {"Foo": 1, "Bar": 2}}` instead of `{"Sticker | Foo": 1, "Sticker | Bar": 2}`) one can save like 200 kb of memory.

I did hope for more :/

Reply:
Ok, to be honest if it's 2 MB of memory or 600 KB is probably irrelevant in this context 😅. The filter speed would be more important.

Reply:
I agree, memory is not as important as speed.

Currently the result sorting may give a lot of unwanted results unfortunately though.

Take a look at these examples:
![image](https://github.com/GODrums/BetterBuff/assets/21990230/1d3676ac-6fb7-40f1-9032-6f948f1d97bc)
![image](https://github.com/GODrums/BetterBuff/assets/21990230/f01d48f7-200a-4048-9efb-98a12b854327)

Reply:
Hm, yes. I've considered modifying the filter so that the keywords you type only match the start of a word. That would solve your first example. 99% of the time, when you search for something, you start typing the beginning of the word and not the middle of it anyway.

Graffiti Example. I think the best order would be: 
Skins < StatTrak Skins < Special Items (Knives & Gloves) < StatTrak Special Items < Stickers < Other (e.g. Graffiti, Music Kits, Agents...)

Reply:
Yes, although this could then mitigate the results when looking for stickers, agents etc. 

I personally would prefer something like a prioritization in the matching.

So for example the earlier a word is matched the higher the prio. So if the first word (e.g. Karambit as knife type) matches, it gets a higher prio then a matching later in the item name (e.g. Karambit as variant of a grafitti). Then just sort all found matches after the assigned prios.

Reply:
Ok I forgot Souvenirs:

Skins < StatTrak Skins < Souvenir Skins < Knives & Gloves < StatTrak Knives < Stickers < Other (e.g. Graffiti, Music Kits, Agents...)

Reply:
> Yes, although this could then mitigate the results when looking for stickers, agents etc.
> 
> I personally would prefer something like a prioritization in the matching.
> 
> So for example the earlier a word is matched the higher the prio. So if the first word (e.g. Karambit as knife type) matches, it gets a higher prio then a matching later in the item name (e.g. Karambit as variant of a grafitti). Additionally we could reduce prios for partial word matches in difference to full-word matches (Exo`skeleton` vs. `Skeleton` Knife). Then just sort all found matches after the assigned prios.

Yes, that was one of the first approaches that came to my mind. However, I thought that **always** evaluating the full 22k different item strings would be too slow in general. That's why I implemented it with the rough "order of importance" being baked into the data itself and then just aborting after the first 6 matches.

Reply:
However, I didn't actually test and implement this other approach yet and your're defiantly right. Evaluating and scoring every item would allow for a much more intelligent search.

I will look into it.

Reply:
Currently the quickest search (aborting pretty much at the beginning (for example, searching for "ak") is like 2ms.
The worst search (searching for "stattrak music kit" which are the very end of the `buff-items-sorted.json`) is around 22ms.

To be honest, 20-30ms still seems to be responsive enough, so maybe always evaluating all 22k items would be worthwhile.

Reply:
Everything below `50ms` is competely fine in the current web standard. Even up to `100ms` can be totally justifyable.

Looking forward to seeing your results 😄